src/Core/OrganizationFeatures/OrganizationSubscriptions/UpdateSecretsManagerSubscriptionCommand.cs

```diff
@@ -74,6 +70,15 @@ public async Task UpdateSecretsManagerSubscription(SecretsManagerSubscriptionUpd
         await SendEmailIfAutoscaleLimitReached(organization);
     }
 
+    public async Task AdjustServiceAccountsAsync(Organization organization, int smServiceAccountsAdjustment)
+    {
+        var secretsManagerSubscriptionUpdate = new SecretsManagerSubscriptionUpdate(
+            organization, seatAdjustment: 0, maxAutoscaleSeats: null,
+            serviceAccountAdjustment: smServiceAccountsAdjustment, maxAutoscaleServiceAccounts: null);
```

Comment:
Good call! I decided on just getting the values from the current `organization` object. I prefer to use the existing logic, adding specific logic for this would require more maintenance.
I also added a unit test for this scenario.